server/src/main/java/com/paypal/selion/grid/servlets/GridAutoUpgradeDelegateServlet.java

```diff
@@ -114,21 +117,34 @@ protected void process(HttpServletRequest request, HttpServletResponse response)
             if (idList == null) {
                 // there were no nodes that failed to auto upgrade
                 for (RemoteProxy eachProxy : this.getRegistry().getAllProxies()) {
-                    SeLionRemoteProxy proxy = (SeLionRemoteProxy) eachProxy;
-                    if (!proxy.release(downloadJSON)) {
-                        pendingProxy.add(proxy.getId());
+                    // TODO :: Address the assumption here that any node which uses SeLionRemoteProxy also has
+                    // NodeAutoUpgradeServlet available. Without truth in this assumption, the call to
+                    // proxy.release() will do nothing.
+                    try {
+                        SeLionRemoteProxy proxy = (SeLionRemoteProxy) eachProxy;
+                        if (!proxy.release(downloadJSON)) {
+                            pendingProxy.add(proxy.getId());
+                        }
+                    } catch (ClassCastException e) {
+                        // ignore, node is not likely compatible with NodeAutoUpgradeServlet
```

Comment:
@mach6 I think we can log the exception, because it will helpful when debugging.